src/path-results.js

```diff
@@ -0,0 +1,806 @@
+    this.toggleHighlighted = this.toggleHighlighted.bind(this);
+    this.soloHighlighted = this.soloHighlighted.bind(this);
+    this.changeSort = this.changeSort.bind(this);
+    this.downloadCsv = this.downloadCsv.bind(this);
+  }
+
+  // when component updates
+  componentDidUpdate(prevProps) {
+    if (this.props.pathQueries !== prevProps.pathQueries) {
+      this.updateAllChecked();
+      this.updateAllHighlighted();
+    }
+  }
+
+  // toggle show more/less
+  toggleShowMore(event) {
+    if (event)
+      event.preventDefault();
+
+    this.setState({ showMore: !this.state.showMore });
+  }
+
+  // updates state whether all rows are checked
+  updateAllChecked() {
+    if (this.allChecked())
+      this.setState({ allChecked: true });
+    else
+      this.setState({ allChecked: false });
+  }
+
+  //
+  toggleAllChecked() {
+    if (this.allChecked())
+      this.uncheckAll();
+    else
+      this.checkAll();
+  }
+
+  // checks whether all paths are checked
+  allChecked() {
+    for (const pathQuery of this.props.pathQueries) {
+      for (const path of pathQuery.paths) {
+        if (!path.checked)
+          return false;
+      }
+    }
+
+    return true;
+  }
+
+  // makes all paths checked
+  checkAll() {
+    const pathQueries = this.props.pathQueries.slice();
+
+    for (const pathQuery of pathQueries) {
+      for (const path of pathQuery.paths)
+        path.checked = true;
+    }
+
+    this.props.dispatch(updatePathQueries({ pathQueries: pathQueries }));
+  }
+
+  // makes all paths unchecked
+  uncheckAll() {
+    const pathQueries = this.props.pathQueries.slice();
+
+    for (const pathQuery of pathQueries) {
+      for (const path of pathQuery.paths)
+        path.checked = false;
+    }
+
+    this.props.dispatch(updatePathQueries({ pathQueries: pathQueries }));
+  }
+
+  // checks whether all paths besides the specified are unchecked
+  allOthersUnchecked(thePath) {
+    for (const pathQuery of this.props.pathQueries) {
+      for (const path of pathQuery.paths) {
+        if (!this.comparePaths(path, thePath) && path.checked)
+          return false;
+      }
+    }
+
+    return true;
+  }
+
+  // toggles the specified paths on/off
+  toggleChecked(thePath) {
+    const pathQueries = this.props.pathQueries.slice();
+
+    for (const pathQuery of pathQueries) {
+      for (const path of pathQuery.paths) {
+        if (this.comparePaths(path, thePath))
+          path.checked = !path.checked;
+      }
+    }
+
+    this.props.dispatch(updatePathQueries({ pathQueries: pathQueries }));
+  }
+
+  // solo path (turn all others off)
+  soloChecked(thePath) {
+    const pathQueries = this.props.pathQueries.slice();
+    const allOthersUnchecked = this.allOthersUnchecked(thePath);
+
+    for (const pathQuery of pathQueries) {
+      for (const path of pathQuery.paths) {
+        if (allOthersUnchecked)
+          path.checked = true;
+        else {
+          if (this.comparePaths(path, thePath))
```

Comment:
I guess lines 182-185 can be as simple as:
```path.checked = this.comparePaths(path, thePath)```

Reply:
I simplified this even further in both `path-results.js` and `metapath-results.js` like so:

```
if (allOthersUn || compare(...))
    path.checked = true;
else
    path.checked = false;
```

In this case, I prefer keeping the if/then rather than doing `checked = (condition1 || condition2)`, just to keep it a little more explicit and readable.